### PR TITLE
Week1 amir

### PR DESCRIPTION
**Pull request by Rashadjaxon**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ x ] Coding Challenges
- [ x ] Coding Concepts
- [ x ] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -35,35 +58,100 @@ const temperature3 = 212
...
-// Expected output: 8
 
 // --------------------5) Create the code that will sort all the numbers in an array from largest to smallest. Use the test variables provided below.
 
-// Pseudo code:
+//Pseudo code:
+
+//Define the arrays sanDiegoSummerTemperatures and sanDiegoWinterTemperatures
+
+//Sort each array in descending order, arranging the numbers from largest to smallest
```
Comment (Genemichael25) [inline code-challenges.js:133]: Great job using the sort method to arrange in the order you need. You should include the function you are passing explaining the a, b in your pseudocode.

File: code-challenges.js [hunk 2]
```diff
@@ -35,35 +58,100 @@ const temperature3 = 212
...
+  // Using a for loop, start at the end of the input string and iterate backwards until the beginning.
+ 
+  // With each iteration, add the current character to the reversed string variable.
+
+  // Return the reversed string variable.
 
 const currentCohort = "Delta 2023"
 // Expected output: "3202 atleD"
 
+const reverseString = (str) => {
```
Comment (Genemichael25) [inline code-challenges.js:95]: Really interesting approach!

File: code-challenges.js [hunk 3]
```diff
@@ -35,35 +58,100 @@ const temperature3 = 212
...
+
+  // Get the length of the combined array using the length
+
+  //  Return the length of the combined array.
+
+
 const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
 const padres1998WorldSeriesRuns = [6, 3, 5, 3]
+const baseball =  padres1984WorldSeriesRuns.concat(padres1988WorldSeriesRuns).length;
+console.log
```
Comment (Genemichael25) [inline code-challenges.js:73]: What are you console logging here?

File: code-challenges.js [hunk 4]
```diff
@@ -22,7 +24,28 @@ const fruit4 = "kiwi"
...
+
+// Evaluate the number with conditionals to determine if it is above, at, or below boiling point.
+
+// Use the provided boiling point temperature of 212 degrees as the threshold for comparison.
+
+// Return a string with the temperature and its relation to boiling point.
+
+const temperatureCheck = (temp) => {
+  if(temp<212) {
+    return "{temp} is above the boiling point"
```
Comment (Genemichael25) [inline code-challenges.js:37]: Nicely done! This is a good start and I think it just needs a small change. Look at your first condition being evaluated and the return. Then you can reference the syllabus for string interpolation syntax and you should be able to refactor this. 👍🏽

File: coding-concepts.js [hunk 5]
```diff
@@ -5,50 +5,60 @@
...
 // --------------------1) What will this log?
 
 const cohort = "LEARN 2023"
 // console.log(cohort.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: 10
+
+// b) Verify and explain: log will output the length of the string
```
Comment (Genemichael25) [inline coding-concepts.js:25]: Exactly! Including the space. Good job!

File: coding-concepts.js [hunk 6]
```diff
@@ -5,50 +5,60 @@
...
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean", "undefined"]
 // console.log(typeof dataTypes.length)
 
 // a) Your answer:
+//"number"
+
 // b) Verify and explain:
+//you use typeof to show a string and using it with .length shows the first string in the index
```
Comment (Genemichael25) [inline coding-concepts.js:63]: Not quite, .length will return a number, in this case the number of elements in the array. typeof then looks at that 4 and recognizes its a number.

File: interview-questions.md [hunk 7]
```diff
@@ -7,51 +7,70 @@ First, without external resources. Challenge yourself to answer from memory as i
...
+//GitHub is a web-based platform for hosting Git repositories and collaborating on code
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: 
+//logical
 
 Researched answer:
+//Equality, Logical, Comparison
```
Comment (Genemichael25) [inline interview-questions.md:33]: This is a great place to expand on your initial answer and provide examples.

File: interview-questions.md [hunk 8]
```diff
@@ -7,51 +7,70 @@ First, without external resources. Challenge yourself to answer from memory as i
...
 
 1. What is the difference between an array's index and its value?
 
 Your answer:
+//an arrays index is a series of characters, value are the positions they're in 
 
 Researched answer:
+//The index is the position of an element in the array, while the value is the data stored at that position.
```
Comment (Genemichael25) [inline interview-questions.md:41]: Nice follow-up and correcting that answer. The index is the position of the element in the array, not the value.

File: interview-questions.md [hunk 9]
```diff
@@ -7,51 +7,70 @@ First, without external resources. Challenge yourself to answer from memory as i
...
+//Arrays and strings are both ordered collections. Arrays can store any type of data and are mutable, while strings store characters only and are immutable
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
 Your answer:
+//take turns, take breaks, communicate. 
+//Drivers do all the typing and Navigators plan out the next steps and assist
 
 Researched answer:
+//Pair programming involves two roles: the driver, who writes the code, and the navigator, who observes, reviews, and guides the driver. They collaborate closely, switching roles frequently to enhance productivity, code quality, and learning.
```
Comment (Genemichael25) [inline interview-questions.md:60]: Collaborating closely and maintaining that communication is key to proper pair programming.

**Review by Genemichael25 — APPROVED** [review]
Amir, this is the first of the assessments and I would like to give you a few things to work on to help with your growth as a developer, as well as, point out some things that you did well. Firstly, I want to stress the importance of timeliness with these assessments. By turning these in late, you risk falling further behind because you don't receive the next week's assessment at the time everyone else does and the feedback that may help you is also delayed. Next, the interview questions are meant to help you think about some questions you may be asked during an interview. I really want you to use the researched portion to help you expand on your initial answer and get comfortable talking about these concepts. You did well with the pseudocode in the coding challenges and explaining most of the coding concepts! This assessment is now complete and I have left comments throughout for your consideration. Please let me know if you have any questions! Thanks

File: code-challenges.js (hunk 3 above)
Comment (Rashadjaxon) [reply code-challenges.js:73]: (combinedArrayLength);

File: code-challenges.js (hunk 1 above)
Comment (Rashadjaxon) [reply code-challenges.js:133]: okay, I'll remember to do that in the future.

**Pull request merged**